codec_fixed.go

```diff
@@ -26,9 +26,9 @@ func createDecoderOfFixed(schema Schema, typ reflect2.Type) ValDecoder {
 			break
 		}
 		switch {
-		case typ.RType() == durRType && ls.Type() == Duration:
+		case typ.Type1().ConvertibleTo(durType) && ls.Type() == Duration:
```

Comment:
It would be more efficient to do the `typ.Type1()` once and reuse it. It is fairly expensive. This goes for all switch checks.